src/components/calendar/TableViewToolbar.tsx

```diff
@@ -0,0 +1,347 @@
+import MomentUtils from '@date-io/moment';
+import { MTableToolbar, Options } from '@material-table/core';
+import {
+  CircularProgress,
+  Grid,
+  makeStyles,
+  TextField,
+} from '@material-ui/core';
+import Autocomplete from '@material-ui/lab/Autocomplete';
+import {
+  KeyboardDatePicker,
+  MuiPickersUtilsProvider,
+} from '@material-ui/pickers';
+import clsx from 'clsx';
+import moment, { Moment } from 'moment';
+import React, { Dispatch, useEffect, useState } from 'react';
+import { useHistory } from 'react-router';
+
+import { Equipment, ScheduledEventFilter } from 'generated/sdk';
+import { useQuery } from 'hooks/common/useQuery';
+import { PartialEquipment } from 'hooks/equipment/useEquipments';
+import { PartialInstrument } from 'hooks/instrument/useUserInstruments';
+import { toTzLessDateTime } from 'utils/date';
+
+import { CalendarScheduledEvent } from './Event';
+import { ToolbarAdditionalProps } from './Toolbar';
+
+const useStyles = makeStyles((theme) => ({
+  flex: {
+    display: 'flex',
+  },
+  tooltip: {
+    margin: theme.spacing(0, 0, 2, 0),
+  },
+  justifyContentFlexEnd: {
+    justifyContent: 'flex-end',
+  },
+  verticalCenter: {
+    alignItems: 'center',
+  },
+  centered: {
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  calendarViewSelect: {
+    minWidth: 120,
+  },
+  smaller: {
+    fontSize: '0.875rem',
+    marginBottom: theme.spacing(1),
+  },
+}));
+
+type TableViewToolbarProps = {
+  onDateRangeChange: (startDate: Moment | null, endDate: Moment | null) => void;
+  startsAtDate: string;
+  endsAtDate: string;
+};
+
+function TableViewToolbar({
+  onDateRangeChange,
+  startsAtDate,
+  endsAtDate,
+  instruments,
+  instrumentsLoading,
+  equipments,
+  equipmentsLoading,
+}: TableViewToolbarProps & ToolbarAdditionalProps) {
+  const query = useQuery();
+  const classes = useStyles();
+  const history = useHistory();
+  const [startsAt, setStartsAt] = useState<Moment | null>(moment(startsAtDate));
+  const [endsAt, setEndsAt] = useState<Moment | null>(moment(endsAtDate));
+  const [queryEquipment, setQueryEquipment] = useState<number[]>([]);
+
+  const queryInstrument = query.get('instrument');
+
+  useEffect(() => {
+    setQueryEquipment(
+      query
+        .get('equipment')
+        ?.split(',')
+        .map((num) => parseInt(num)) || []
+    );
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
I'm not completely sure but is the `useEffect` necessary here?
Try plugging in the value from the useQuery hook directly into the `queryEquipment`, as an initial value

Reply:
I understand your point but if it is not inside `useEffect` it gives a warning: The `queryEquipment` logical expression could make the dependencies of `useEffect` Hook (at line 124) change on every render. Move it inside the `useEffect` callback. Alternatively, wrap the initialization of `queryEquipment` in its own `useMemo()` Hook. I guess I can use one of the options with `useEffect` like it is now or wrap it with `useMemo`.

Reply:
Let me know if you think `useMemo` is better than this current solution.